openff/interchange/_experimental.py

```diff
@@ -0,0 +1,25 @@
+import os
+
+from openff.interchange.exceptions import ExperimentalFeatureException
+
+
+def experimental(func):
+    """
+    Decorate a function as experimental, requiring environment variable opt-in.
+
+    To use the wrapped function, set the environment variable INTERCHANGE_EXPERIMENTAL=1.
+    """
+
+    def wrapper(*args, **kwargs):
```

Comment:
The `functools.wraps` decorator tells Python that a function is a wrapper around some other function, which preserves the signature and docstring and so on. PyRight seems to be able to figure it out on its own, but Sphinx couldn't.

We can also dynamically modify the docstring to add a test-tube symbol and a warning (or whatever). The call to `cleandoc` is important in case the docstring is indented (as almost all docstrings are) or starts with a newline.

```suggestion
from functools import wraps
from inspect import cleandoc
import os

from openff.interchange.exceptions import ExperimentalFeatureException


def experimental(func):
    """
    Decorate a function as experimental, requiring environment variable opt-in.

    To use the wrapped function, set the environment variable INTERCHANGE_EXPERIMENTAL=1.
    """

    firstline, _, remainder = cleandoc(func.__doc__).partition("\n")

    func.__doc__ = (
        "🧪 "
        + firstline
        + "\n\n.. admonition:: Experimental\n\n    This object is experimental and should not be used in production.\n"
        + remainder
    )

    @wraps(func)
    def wrapper(*args, **kwargs):
```